projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { Order, OrderApprovalService, RoutingService } from '@spartacus/core';
+import { Observable } from 'rxjs';
+import {
+  filter,
+  map,
+  pluck,
+  shareReplay,
+  switchMap,
+  tap,
+} from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OrderApprovalDetailService {
+  protected approvalCode$ = this.routingService
+    .getRouterState()
+    .pipe(map((routingData) => routingData.state.params.approvalCode));
+
+  protected orderData$ = this.approvalCode$.pipe(
+    filter(Boolean),
+    tap((approvalCode: string) =>
+      this.orderApprovalService.loadOrderApproval(approvalCode)
+    ),
+    switchMap((approvalCode: string) =>
+      this.orderApprovalService.get(approvalCode)
+    ),
+    pluck('order'),
```

Comment:
I see the headscratching part due usage of different api and what needed to be done to re-use existing components.

But now that I think of it, we would use this service to use other data from the response, so I was thinking maybe one day we will have redundancies even though it is not really one since this is a must for re-using.

Reply:
nvm this was addressed in the form component :)